tsensor/viz.py

```diff
@@ -71,6 +91,62 @@ def __init__(self, statement, fontname, fontsize, dimfontname, dimfontsize,
         self.offending_expr = None
         self.fignumber = None
 
+    def get_shape_color(self, shape):
+        """Get color based on shape. Current logic distinguishes vector/N-d tensor."""
+        if shape == 1:
+            return self.vectorcolor
+        elif shape > 1:
+            return self.matrixcolor
+        else:
+            raise ValueError("Empty shape unsupported.")
+
+    @staticmethod
+    def _split_dtype_precision(s):
+        """Split the final integer part from a string"""
+        head = s.rstrip('0123456789')
+        tail = s[len(head):]
+        return head, tail
+
+    def _get_dtype_color_indices(self, dtype_name, dtype_precision):
+        if dtype_name not in self._dtype_name2color:
```

Comment:
it looks like base color can change for same type depending on order in which this function is called, right?

Reply:
I'll assign a specific color like "greenish" to floats or whatever.

Reply:
BTW, i'm altering in a copy of your branch so i'll take a whack at fixing. I've made some changes like "no old style shape colors".

Reply:
Ok I'm thinking we specify a fixed color per type then bit size is gradation for 4, 8, 16, 32, 64, 128 (auto computed with color map).  I'm looking at types for various packages.